test/jasmine/tests/axes_test.js

```diff
@@ -5451,22 +5552,414 @@ describe('Test axes', function() {
                         '2020-01-01 21:00',
                         '2020-01-02'
                     ], [
-                        ['', 'Wed-0', 'Wed-0', 'Wed-0', 'Wed-0', 'Wed-0', 'Wed-0', 'Wed-0', 'Wed-0', 'Thu-0'],
-                        ['', 'Wed-000', 'Wed-000', 'Wed-000', 'Wed-000', 'Wed-000', 'Wed-000', 'Wed-000', 'Wed-000', 'Thu-000'],
                         ['', 'Wed-1577836800000', 'Wed-1577847600000', 'Wed-1577858400000', 'Wed-1577869200000', 'Wed-1577880000000', 'Wed-1577890800000', 'Wed-1577901600000', 'Wed-1577912400000', 'Thu-1577923200000'],
                         ['', 'Wed-1577836800', 'Wed-1577847600', 'Wed-1577858400', 'Wed-1577869200', 'Wed-1577880000', 'Wed-1577890800', 'Wed-1577901600', 'Wed-1577912400', 'Thu-1577923200'],
-                        ['', 'Wed-00', 'Wed-00', 'Wed-00', 'Wed-00', 'Wed-00', 'Wed-00', 'Wed-00', 'Wed-00', 'Thu-00'],
-                        ['', 'Wed-00', 'Wed-00', 'Wed-00', 'Wed-00', 'Wed-00', 'Wed-00', 'Wed-00', 'Wed-00', 'Thu-00'],
                         ['', 'Wed-00', 'Wed-03', 'Wed-06', 'Wed-09', 'Wed-12', 'Wed-15', 'Wed-18', 'Wed-21', 'Thu-00'],
                         ['', 'Wed-12', 'Wed-03', 'Wed-06', 'Wed-09', 'Wed-12', 'Wed-03', 'Wed-06', 'Wed-09', 'Thu-12'],
-                        ['', 'Wed-AM', 'Wed-AM', 'Wed-AM', 'Wed-AM', 'Wed-PM', 'Wed-PM', 'Wed-PM', 'Wed-PM', 'Thu-AM'],
                         ['', 'Wed-00:00:00', 'Wed-03:00:00', 'Wed-06:00:00', 'Wed-09:00:00', 'Wed-12:00:00', 'Wed-15:00:00', 'Wed-18:00:00', 'Wed-21:00:00', 'Thu-00:00:00']
                     ][i]);
                 })
                 .catch(failTest)
                 .then(done);
             });
         });
+
+
+        [
+            {
+                formatter: '%p',
```

Comment:
@alexcjohnson @nicolaskruchten 
I was wondering maybe we should re-position this AM/PM case using `%p`. What do you think?

Reply:
Also for hours (i.e using `%H` or `%I`) I feel we could move the labels to the middle of period.

Reply:
I made two adjustments in https://github.com/plotly/plotly.js/pull/5089/commits/12ce0665525f60e8731079ff856a501e1ac15b2d & in https://github.com/plotly/plotly.js/pull/5089/commits/f47c8206b7f36b1f6faed0d8437335447479faf3 to cover those cases.